docs/dev.md

```diff
@@ -0,0 +1,54 @@
+# Development guide
+
+## Cloning the repository
+
+To be able to run the sanity tests, the recommended way to extract the repository is like this:
+
+```ShellSession
+$ cd <your workspace>
+$ mkdir -p ansible_collections/redhatci
+$ cd ansible_collections/redhatci
+$ git clone git@github.com:redhatci/ansible-collection-redhatci-ocp.git ocp
+$ cd ocp
+...
+```
+
+## Running the sanity tests locally
+
+The sanity tests are run using the [`ansible-test`](https://docs.ansible.com/ansible/latest/dev_guide/testing_sanity.html) tool. It comes from the `ansible-core` package. This is what is run in [the CI pipeline](https://github.com/redhatci/ansible-collection-redhatci-ocp/blob/main/.github/workflows/pr.yml#L40).
+
+```ShellSession
+$ ansible-test sanity --verbose --docker --color --coverage --failure-ok
+```
+
+Legacy code is expected to fail some sanity tests. The ask is to not introduce new failures and fix the errors in the legacy code when you are modifying it. It is an iterative process to clean up the legacy code.
+
+## Pull Requests
+
+The collection is using the [GitHub Flow](https://guides.github.com/introduction/flow/) for the development. The main branch is `main`. The `main` branch is protected and requires a PR to be merged. The PRs are reviewed by the maintainers of the collection.
+
+Keep the PRs small and focused on a single topic. It is easier to review and merge. It also helps to keep the git history clean. If you need to make a change to a PR, use the `git commit --amend` and `git push --force` to update the PR. It will update the PR automatically. Squash the commits when you are ready to merge the PR.
```

Comment:
It would be good to add in here that any "work in progress" PR can be set as "Draft", what about:

PRs can be created as a [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests), use this when the change is not yet ready for review or set a PR to Draft if it needs more time before continuong with the review process

Reply:
That's not the case anymore. Now when there is a force push, you can see the change with the previous version.

Ok for the draft.

Reply:
I have added alternative options to show the 2 ways to do it. Let me know if this is fine.